test/libsolidity/smtCheckerTests/try_catch/try_1.sol

```diff
@@ -0,0 +1,22 @@
+pragma experimental SMTChecker;
+contract C {
+	int x;
+
+	function g() public  {
+		x = 42;
+	}
+
+	function f() public {
+		x = 0;
+		bool success = false;
+		try this.g() {
+			success = true;
+		} catch (bytes memory s) {
+			assert(x == 0); // should hold
+		}
+		assert(success); // fails for now, since external call is over-approximated (both success and fail are considered possible) for now even for known code
```

Comment:
If we use `CHC::externalFunctionCall` shouldn't it call the trustedExternalCall code?

Reply:
It does. What I mean by this comment that when analyzing the control flow, we still add the edges to all catch blocks, although maybe some of the edges cannot be taken?

This is related to a question I had. Can an external call like this one, or an external call to public getter fail? My intuition is that it still can fail, no?

Reply:
Right, I think it can. For example for an array getter, if you give an index out of bonuds